d2/src/main/java/com/linkedin/d2/jmx/RelativeLoadBalancerStrategyJmx.java

```diff
@@ -0,0 +1,189 @@
+import com.linkedin.d2.balancer.strategies.relative.RelativeLoadBalancerStrategy;
+import com.linkedin.d2.balancer.strategies.relative.StateUpdater;
+import com.linkedin.d2.balancer.strategies.relative.TrackerClientState;
+import com.linkedin.d2.balancer.util.partitions.DefaultPartitionAccessor;
+import com.linkedin.util.degrader.CallTracker;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class RelativeLoadBalancerStrategyJmx implements RelativeLoadBalancerStrategyJmxMBean
+{
+  private final RelativeLoadBalancerStrategy _strategy;
+
+  public RelativeLoadBalancerStrategyJmx(RelativeLoadBalancerStrategy strategy)
+  {
+    _strategy = strategy;
+  }
+
+  @Override
+  public double getLatencyStandardDeviation()
+  {
+    Map<TrackerClient, TrackerClientState> stateMap =
+        _strategy.getPartitionState(DefaultPartitionAccessor.DEFAULT_PARTITION_ID).getTrackerClientStateMap();
+
+    return calculateStandardDeviation(stateMap.keySet());
+  }
+
+  @Override
+  public double getLatencyMeanAbsoluteDeviation()
+  {
+    Map<TrackerClient, TrackerClientState> stateMap =
+        _strategy.getPartitionState(DefaultPartitionAccessor.DEFAULT_PARTITION_ID).getTrackerClientStateMap();
+
+    double avgLatency = getAvgClusterLatency(stateMap.keySet());
+
+    return stateMap.keySet().stream()
+        .map(trackerClient -> Math.abs(StateUpdater.getAvgHostLatency(trackerClient.getCallTracker().getCallStats()) - avgLatency))
+        .mapToDouble(Double::doubleValue)
+        .average()
+        .orElse(0);
+  }
+
+  @Override
+  public double getAboveAverageLatencyStandardDeviation()
+  {
+    Map<TrackerClient, TrackerClientState> stateMap =
+        _strategy.getPartitionState(DefaultPartitionAccessor.DEFAULT_PARTITION_ID).getTrackerClientStateMap();
+
+    double avgLatency = getAvgClusterLatency(stateMap.keySet());
+
+    Set<TrackerClient> aboveAvgClients = stateMap.keySet().stream()
+        .filter(trackerClient -> StateUpdater.getAvgHostLatency(trackerClient.getCallTracker().getCallStats()) > avgLatency)
+        .collect(Collectors.toSet());
+
+    return calculateStandardDeviation(aboveAvgClients);
+  }
+
+  @Override
+  public double getMaxLatencyRelativeFactor()
+  {
+    Map<TrackerClient, TrackerClientState> stateMap =
+        _strategy.getPartitionState(DefaultPartitionAccessor.DEFAULT_PARTITION_ID).getTrackerClientStateMap();
+
+    double avgLatency = getAvgClusterLatency(stateMap.keySet());
+    long maxLatency = stateMap.keySet().stream()
+        .map(trackerClient -> StateUpdater.getAvgHostLatency(trackerClient.getCallTracker().getCallStats()))
+        .mapToLong(Long::longValue)
+        .max()
+        .orElse(0L);
+
+    return avgLatency == 0 ? 0 : maxLatency / avgLatency;
+  }
+
+  @Override
+  public double getNthPercentileLatencyRelativeFactor(double pct)
```

Comment:
Wondering how do you plan to implement this in container? Do you plan to go with multiple percentile relative factors?

Reply:
I'm not sure about which percentile to use. I can start by trying out p99, p95. So having a more general interface here might be easier for later experimentations.